plugins/tweak/tweaks/material-size-for-melting.h

```diff
@@ -0,0 +1,121 @@
+        return get_material_size_for_melting(this, INTERPOSE_NEXT(getMaterialSizeForMelting)());
+    }
+};
+IMPLEMENT_VMETHOD_INTERPOSE(material_size_for_melting_armor_hook, getMaterialSizeForMelting);
+
+struct material_size_for_melting_gloves_hook : df::item_glovesst {
+    typedef df::item_glovesst interpose_base;
+
+    DEFINE_VMETHOD_INTERPOSE(int32_t, getMaterialSizeForMelting, ()) {
+        return get_material_size_for_melting(this, INTERPOSE_NEXT(getMaterialSizeForMelting)(), 2.0f);
+    }
+};
+IMPLEMENT_VMETHOD_INTERPOSE(material_size_for_melting_gloves_hook, getMaterialSizeForMelting);
+
+struct material_size_for_melting_shoes_hook : df::item_shoesst {
+    typedef df::item_shoesst interpose_base;
+
+    DEFINE_VMETHOD_INTERPOSE(int32_t, getMaterialSizeForMelting, ()) {
+        return get_material_size_for_melting(this, INTERPOSE_NEXT(getMaterialSizeForMelting)(), 2.0f);
+    }
+};
+IMPLEMENT_VMETHOD_INTERPOSE(material_size_for_melting_shoes_hook, getMaterialSizeForMelting);
+
+struct material_size_for_melting_helm_hook : df::item_helmst {
+    typedef df::item_helmst interpose_base;
+
+    DEFINE_VMETHOD_INTERPOSE(int32_t, getMaterialSizeForMelting, ()) {
+        return get_material_size_for_melting(this, INTERPOSE_NEXT(getMaterialSizeForMelting)());
+    }
+};
+IMPLEMENT_VMETHOD_INTERPOSE(material_size_for_melting_helm_hook, getMaterialSizeForMelting);
+
+struct material_size_for_melting_pants_hook : df::item_pantsst {
+    typedef df::item_pantsst interpose_base;
+
+    DEFINE_VMETHOD_INTERPOSE(int32_t, getMaterialSizeForMelting, ()) {
+        return get_material_size_for_melting(this, INTERPOSE_NEXT(getMaterialSizeForMelting)());
+    }
+};
+IMPLEMENT_VMETHOD_INTERPOSE(material_size_for_melting_pants_hook, getMaterialSizeForMelting);
+
+struct material_size_for_melting_weapon_hook : df::item_weaponst {
+    typedef df::item_weaponst interpose_base;
+
+    DEFINE_VMETHOD_INTERPOSE(int32_t, getMaterialSizeForMelting, ()) {
+        return get_material_size_for_melting(this, INTERPOSE_NEXT(getMaterialSizeForMelting)());
+    }
+};
+IMPLEMENT_VMETHOD_INTERPOSE(material_size_for_melting_weapon_hook, getMaterialSizeForMelting);
+
+struct material_size_for_melting_trapcomp_hook : df::item_trapcompst {
+    typedef df::item_trapcompst interpose_base;
+
+    DEFINE_VMETHOD_INTERPOSE(int32_t, getMaterialSizeForMelting, ()) {
+        return get_material_size_for_melting(this, INTERPOSE_NEXT(getMaterialSizeForMelting)());
+    }
+};
+IMPLEMENT_VMETHOD_INTERPOSE(material_size_for_melting_trapcomp_hook, getMaterialSizeForMelting);
+
+struct material_size_for_melting_tool_hook : df::item_toolst {
+    typedef df::item_toolst interpose_base;
+
+    DEFINE_VMETHOD_INTERPOSE(int32_t, getMaterialSizeForMelting, ()) {
+        return get_material_size_for_melting(this, INTERPOSE_NEXT(getMaterialSizeForMelting)());
+    }
+};
+IMPLEMENT_VMETHOD_INTERPOSE(material_size_for_melting_tool_hook, getMaterialSizeForMelting);
```

Comment:
This is extremely repetitive. I suggest to have a look at `partial-items.h` to see how to reduce the per-structure part to a single line per structure .

Reply:
Thanks, done.